File: PROJET_ADOPTE/src/main/java/fr/formation/model/Utilisateur.java
package fr.formation.model;

import java.time.LocalDate;
import java.util.List;
import java.util.Set;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.JoinTable;
import javax.persistence.ManyToMany;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.Table;

import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonView;


@Entity
@Table(name = "utilisateur")
public class Utilisateur {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "uti_id")
	@JsonView(JsonViews.Common.class)
	private int id;
	
	@Column(name = "uti_date_naissance",nullable=false)
	@JsonView(JsonViews.Common.class)
	private LocalDate dateNaissance;
	
	@Column(name = "uti_pseudo", length = 100, nullable = false)
	@JsonView(JsonViews.Common.class)
	private String pseudo;
	@Column(name = "uti_nom", length = 100, nullable = false)
	@JsonView(JsonViews.Common.class)
	private String nom;
	@Column(name = "uti_prenom", length = 100, nullable = false)
	@JsonView(JsonViews.Common.class)
	private String prenom;
	@Column(name = "uti_mail", length = 150, nullable = true)
	@JsonView(JsonViews.Common.class)
	private String mail;
	@Column(name = "uti_mdp", length = 100, nullable = false)
	@JsonView(JsonViews.Common.class)
	private String mdp;
	@Column(name = "uti_adresse", length = 100, nullable = false)
	@JsonView(JsonViews.Common.class)
	private String adresse;
	@Column(name = "uti_tel", length = 20,nullable=false)
	@JsonView(JsonViews.Common.class)
	private String telephone;
	
	@Enumerated(EnumType.ORDINAL)
	@Column(name = "uti_niveau", nullable = false)
	@JsonView(JsonViews.Common.class)
	private Niveau niveau;
	
	@JsonIgnore
	@ManyToOne//(fetchtype.lazy)
	@JoinColumn(name = "uti_image_id")


	@JsonView(JsonViews.Common.class)
	private Image photoProfil;

	//pas besoin d'attribut ???ge puisqu'on a la date de naissance
//	@Column(name = "uti_age")
//	private int age;
	
//	@JsonIgnore
	@ManyToMany (cascade = CascadeType.PERSIST)
	@JoinTable (name ="style_utilisateur",
	joinColumns=@JoinColumn(name="styuti_utilisateur_id"),
	inverseJoinColumns=@JoinColumn(name="styuti_stylemusical_id"))
	@OnDelete(action = OnDeleteAction.CASCADE)
	@JsonView({JsonViews.UtilisateurAvecStyle.class,
		JsonViews.UtilisateurAvecTout.class})
	private Set<StyleMusical> stylemusical;
	
//	@JsonIgnore
	@ManyToMany (cascade = CascadeType.PERSIST)
	@JoinTable (name ="utilisateur_instrument",
	joinColumns=@JoinColumn(name="utiins_utilisateur_id"),
	inverseJoinColumns=@JoinColumn(name="utiins_instrument_id"))
	@OnDelete(action = OnDeleteAction.CASCADE)
	@JsonView({JsonViews.UtilisateurAvecInstrument.class,
			JsonViews.UtilisateurAvecTout.class})
	private Set<Instrument> listeinstrument;
	

//	@JsonIgnore
	@OneToMany(mappedBy ="createur")
	@JsonView({JsonViews.UtilisateurAvecSon.class,
			JsonViews.UtilisateurAvecTout.class})
	private List<Son> son; // ?? voir avec J??r??my

	
	
	
	public LocalDate getDateNaissance() {
		return dateNaissance;
	}

	public void setDateNaissance(LocalDate dateNaissance) {
		this.dateNaissance = dateNaissance;
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getPseudo() {
		return pseudo;
	}

	public void setPseudo(String pseudo) {
		this.pseudo = pseudo;
	}

	public String getNom() {
		return nom;
	}

	public void setNom(String nom) {
		this.nom = nom;
	}

	public String getPrenom() {
		return prenom;
	}

	public void setPrenom(String prenom) {
		this.prenom = prenom;
	}

	public String getMail() {
		return mail;
	}

	public void setMail(String mail) {
		this.mail = mail;
	}

	public String getMdp() {
		return mdp;
	}

	public void setMdp(String mdp) {
		this.mdp = mdp;
	}

	public String getAdresse() {
		return adresse;
	}

	public void setAdresse(String adresse) {
		this.adresse = adresse;
	}

	public String getTelephone() {
		return telephone;
	}

	public void setTelephone(String telephone) {
		this.telephone = telephone;
	}

	public Niveau getNiveau() {
		return niveau;
	}

	public void setNiveau(Niveau niveau) {
		this.niveau = niveau;
	}

	public Image getPhotoprofil() {
		return photoProfil;
	}

	public void setPhotoprofil(Image photoProfil) {
		this.photoProfil = photoProfil;
	}

	public Set<StyleMusical> getStylemusical() {
		return stylemusical;
	}

	public void setStylemusical(Set<StyleMusical> stylemusical) {
		this.stylemusical = stylemusical;
	}

	public Set<Instrument> getListeinstrument() {
		return listeinstrument;
	}

	public void setListeinstrument(Set<Instrument> listeinstruments) {
		this.listeinstrument = listeinstruments;
	}

	public List<Son> getSon() {
		return son;
	}

	public void setSon(List<Son> son) {
		this.son = son;
	}

	public boolean friendWith(Utilisateur people, Utilisateur chef) {
		// TODO Auto-generated method stub
		return false;
	}
	
}

File: angularAdopte/src/app/model/utilisateur.ts
import { StyleMusical } from './style-musical';
import { Instrument } from './instrument';
import { Compte } from './compte';
import { Image } from './image';

export class Utilisateur extends Compte {
  public get stylemusical(): StyleMusical[] | undefined {
    return this._stylemusical;
  }
  public set stylemusical(value: StyleMusical[] | undefined) {
    this._stylemusical = value;
  }
  public get photoProfil(): Image | undefined {
    return this._photoProfil;
  }
  public set photoProfil(value: Image | undefined) {
    this._photoProfil = value;
  }
  public get listeinstrument(): Instrument[] | undefined {
    return this._listeinstrument;
  }
  public set listeinstrument(value: Instrument[] | undefined) {
    this._listeinstrument = value;
  }
  public get dateNaissance(): string | undefined {
    return this._dateNaissance;
  }
  public set dateNaissance(value: string | undefined) {
    this._dateNaissance = value;
  }
  public get adresse(): string | undefined {
    return this._adresse;
  }
  public set adresse(value: string | undefined) {
    this._adresse = value;
  }
  public get niveau(): string | undefined {
    return this._niveau;
  }
  public set niveau(value: string | undefined) {
    this._niveau = value;
  }
  public get telephone(): number | undefined {
    return this._telephone;
  }
  public set telephone(value: number | undefined) {
    this._telephone = value;
  }
  public get mail(): string | undefined {
    return this._mail;
  }
  public set mail(value: string | undefined) {
    this._mail = value;
  }
  public get prenom(): string | undefined {
    return this._prenom;
  }
  public set prenom(value: string | undefined) {
    this._prenom = value;
  }
  public get nom(): string | undefined {
    return this._nom;
  }
  public set nom(value: string | undefined) {
    this._nom = value;
  }

  constructor(
    _id: number,
    _username: string,
    _password: string,
    private _nom?: string,
    private _prenom?: string,
    private _mail?: string,
    private _telephone?: number,
    private _niveau?: string,
    private _adresse?: string,
    private _dateNaissance?: string,
    private _listeinstrument?: Instrument[],
    private _photoProfil?: Image,
    private _stylemusical?: StyleMusical[]
  ) {
    super(_id, _username, _password);
  }
}
